### ISSUE TITLE
`*-line-out-of-listing` rules incorrectly count lines in source files ending with a newline

### ISSUE THREAD

**Issue by gemshrine** [objectionary/lints#1358]
The three `*-line-out-of-listing` rules compute the listing's length by counting the pieces `tokenize` returns, and a source file ending with a newline yields one piece more than it has lines. So each rule's bound is one too high and a line exactly one past the end of the file is never reported.

The same expression is in all three, on line 9 of each:

```xml
<xsl:variable name="max" select="count(tokenize(/object/listing, '&#10;'))"/>
```

`lines/error-line-out-of-listing.xsl`, `lines/meta-line-out-of-listing.xsl` and `lines/object-line-out-of-listing.xsl`, with the check on line 12 in each:

```xml
<xsl:for-each select="/object/errors/error[number(@line) and @line &gt; $max]">
```

`tokenize("a\nb\n", "\n")` gives `("a", "b", "")` — three items for two lines. EO sources end with a newline, so the trailing empty piece is always there. Measured on a parsed XMIR from `eo-runtime`:

```
listing tail                         : '    a false &gt; closed\n'
count(tokenize(listing, LF))  ($max) : 233
real line count of bool.eo           : 232
```

An `@line` of 233 on that file is out of the listing and passes the `@line > $max` test unreported. The rules do catch anything further out, so the effect is narrow — exactly one line — but it is the most likely wrong value: a line number produced by counting from the wrong end, or by adding one to the last line, lands precisely there.

Subtracting the trailing piece fixes all three:

```xml
<xsl:variable name="max" select="count(tokenize(/object/listing, '&#10;')[. != '' or position() != last()])"/>
```

or, more plainly, by tokenizing the listing with its trailing newline removed:

```xml
<xsl:variable name="max" select="count(tokenize(replace(/object/listing, '&#10;$', ''), '&#10;'))"/>
```

The second form also behaves correctly for a listing that does not end with a newline, where the current expression is already right and must not lose a line.

Worth checking the empty case while the expression is being touched: for an empty listing, `tokenize('', '\n')` returns one empty piece, so `$max` is 1 and a defect on line 1 of a file with no content goes unreported.

**Comment by 0crat** [comment]
@gemshrine Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+3626](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
